File: src/abc049_c.rb
s = gets.chomp

if s =~/^(dreamer|eraser|dream|erase)+$/
  puts "YES"
else
  puts "NO"
end

#words = ["dreamer", "eraser", "dream", "erase"];
#
#while s.length > 0 do
#  words.each do |w|
#    if s =~ /^#{w}/
#      s = s.slice(w.length, s.length - w.length)
#      puts s
#      break
#    end
#
#  end
#
#  if s.length < "dream".length
#    break
#  end
#end
#
#puts s.length == 0 ? "YES" : "NO"

File: src/abc049_c.rs
extern crate regex;
use regex::Regex;
use std::io;

fn main() {

    let s = get_input();
    let re = Regex::new("(dreamer|eraser|dream|erase)+").unwrap();

    if re.is_match(&s) {
        println!("YES");
    } else {
        println!("NO");
    }
}


fn get_input() -> String {
    let mut input_str = String::new();
    io::stdin().read_line(&mut input_str)
        .expect("Failed to read line")
        ;
    input_str
}
